tests/e2e/helm_test.go

```diff
@@ -0,0 +1,167 @@
+	UpdatedTime string            `json:"updated" yaml:"updated"` // Note: not parsable implicitly.
+	Status      HelmReleaseStatus `json:"status" yaml:"status"`
+	Chart       string            `json:"chart" yaml:"chart"`
+	AppVersion  string            `json:"app_version" yaml:"app_version"`
+}
+
+// installHelmChart deploys a Helm chart to the specified kubectl context and
+// namespace using the specified infos, extra arguments can be any of the helm
+// CLI install flag arguments, flag keys and values must be provided separately.
+func installHelmChart(
+	kubectlOptions *k8s.KubectlOptions,
+	helmRepository string,
+	helmChartNameOrLocalPath string,
+	helmChartVersion string,
+	helmReleaseName string,
+	setValues map[string]string,
+) {
+	By(
+		fmt.Sprintf(
+			"Installing Helm chart %s from %s with version %s by name %s",
+			helmChartNameOrLocalPath, helmRepository, helmChartVersion, helmReleaseName,
+		),
+	)
+
+	fixedArguments := []string{
+		"--create-namespace",
+		"--atomic",
+		"--debug",
+	}
+
+	if helmRepository != "" {
+		fixedArguments = append([]string{"--repo", helmRepository}, fixedArguments...)
+	}
+
+	helm.Install(
+		GinkgoT(),
+		&helm.Options{
+			SetValues:      setValues,
+			KubectlOptions: kubectlOptions,
+			Version:        helmChartVersion,
+			ExtraArgs: map[string][]string{
+				"install": fixedArguments,
+			},
+		},
+		helmChartNameOrLocalPath,
+		helmReleaseName,
+	)
+}
+
+// installHelmChart checks whether the specified named Helm release exists in
+// the provided kubectl context and namespace, logs it if it does and returns or
+// alternatively deploys the Helm chart to the specified kubectl context and
+// namespace using the specified infos, extra arguments can be any of the helm
+// CLI install flag arguments, flag keys and values must be provided separately.
+func installHelmChartIfDoesNotExist(
+	kubectlOptions *k8s.KubectlOptions,
+	helmRepository string,
+	helmChartNameOrLocalPath string,
+	helmChartVersion string,
+	helmReleaseName string,
+	setValues map[string]string,
+) {
+	By(fmt.Sprintf("Checking for existing Helm release named %s", helmReleaseName))
+	helmRelease, isInstalled := lookUpInstalledHelmReleaseByName(kubectlOptions, helmReleaseName)
+
+	if isInstalled {
+		By(fmt.Sprintf(
+			"Skipping the installation of %s, because the Helm release is already present: %+v",
+			helmReleaseName, helmRelease,
+		))
+
+		return
+	}
+
+	installHelmChart(
+		kubectlOptions,
+		helmRepository,
+		helmChartNameOrLocalPath,
+		helmChartVersion,
+		helmReleaseName,
+		setValues,
+	)
+}
+
+// listHelmReleases returns a slice of Helm releases retrieved from the cluster
+// using the specified kubectl context and namespace.
+func listHelmReleases(kubectlOptions *k8s.KubectlOptions) []*HelmRelease {
+	By("Listing Helm releases")
+	output, err := helm.RunHelmCommandAndGetOutputE(
+		GinkgoT(),
+		&helm.Options{
+			KubectlOptions: kubectlOptions,
+		},
+		"list",
+		"--output", "yaml",
```

Comment:
We can go with json and later json.Unmarshal([]byte(output), &releases)
yaml unmarshaller converts first the output to json and after into struct

Reply:
I just picked a random format I could parse into an object, I prefer YAML, because in printed format it's more readable to a human eye than JSON for debugging purposes.
For tests the performance difference is immeasurable - the test steps will take significantly more time than the helm list command.
But I can convert it to JSON.

Reply:
I agree with you that for debugging purposes we should use yaml.

Reply:
Fixed in a8def0d.

Reply:
Fixed the fix in 1f98ba9.